eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,144 @@
+param(
+  [string] $InputJsonPath = $GenerateInput,
+  [string] $OutputJsonPath = $GenerateOutput,
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+$inputFilePaths = $inputFileContent.changedFiles;
+$inputFilePaths += $inputFileContent.relatedReadmeMdFiles;
+$inputFilePaths = $inputFilePaths | Select-Object -Unique
+
+$changedFilePaths = $inputFilePaths -join "`n";
+Write-Host "List of changed swagger files and related readmes:`n$changedFilePaths`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($inputFilePath in $inputFilePaths) {
+  foreach ($path in $autorestFilesPath) {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $updatedLines = foreach ($line in $fileContent) {
+      $regexForMatchingShaAndPath = "[\/][0-9a-f]{4,40}[\/]$inputFilePath"
```

Comment:
`$inputFilePath` is going to have backslashes, periods, and other special regex characters that should be escaped or it won't reliably match. Pass it first to `[System.Text.RegularExpressions.Regex]::Escape([string])`: https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.escape?view=net-5.0